### ISSUE TITLE
no-raw-text doesn't work for styled components

### ISSUE THREAD

**Issue by vjsingh** [Intellicode/eslint-plugin-react-native#248]
Unless I'm doing something wrong, the no-raw-text rule doesn't seem to work when using styled-components. This makes it largely useless I think? Would be nice to be able to use this rule but not sure how to do so when using styled-components.

```
import styled from 'styled-components/native';

let view = () => (
  <MyText>This doesn't work</MyText>
);

const MyText = styled(Text)`
  font-size: 16;
`;
```

**Comment by LauraBeatris** [comment]
Same problem... My codebase is using a lot of Typograph component made with ``styled-components`` and unfortunately, I had to disable this rule

**Comment by adibas03** [comment]
+1

**Comment by adibas03** [comment]
First, @vjsingh You should rename the issue to 'no-raw-text doesn't work for `custom` styled components', as the rule actually works, but not for custom components.

Been doing some investigation into what it would take to implement this.
At present, I am yet to get a direct reference to the type of StyledComponent from within the babel nodes which are passed into the rule checking function; only the Customized name of the component show up, so far.
If anyone has some pointers or comment, that could help, I'd appreciate, so I can set up a PR, to fix this, once and for all

**Comment by mattpetrie** [comment]
Also running into this. One possible option to fix it would be to change the `skip` option to accept a regex instead of (or in addition to) an array of strings, so that you could do, for example:
```js
'react-native/no-raw-text': ['error', {skip: '/Text$/'}]
```
to skip any component that is suffixed with `Text`.

**Comment by adibas03** [comment]
@mattpetrie 
That is actually a valid suggestion, we just need to consider the possible loopholes or drawbacks
I was attempting to identify the styled -component component type, but not been able to get it from the babel generated resource for the lint checks

**Comment by fredlemieux** [comment]
> ```js
> /Text$/
> ```

This does not work for me.  I.e. I have a styled component called <TextSummary> if I add skip: ['TextSummary'] the error goes away so the rule is working. But /Text$/ doesn't work, nor does variations on this, ie. /Text.*/   /.*Text.*./ 
Any suggestions?

**Comment by ManAnRuck** [comment]
**RegEx does not work** for me with version: v3.10.0
String does work

**Comment by bombillazo** [comment]
Regex isnt implemented, @mattpetrie was simply giving a suggestion.
